pkg/pipelines/secrets/generate_secret.go

```diff
@@ -0,0 +1,22 @@
+package secrets
+
+import (
+	"crypto/rand"
+	"fmt"
+)
+
+const charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789$:#"
+
+// GenerateSecureString creates a random secret of the desired length specifically for the webhook secrets.
+func GenerateSecureString(length int) (string, error) {
+	b := make([]byte, length)
+	_, err := rand.Read(b)
+	if err != nil {
+		return "", fmt.Errorf("Failed to generate secure webhook secret")
```

Comment:
I wouldn't mention "secure webhook" in the error because this method can serve other purposes than creating secure for webhook.

Reply:
Changes made to the more generic error message.